crates/sui-core/src/authority_server.rs

```diff
@@ -824,36 +804,12 @@ macro_rules! handle_with_decoration {
             }
         }
 
-        let proxy_ip: Option<SocketAddr> =
-            if let Some(op) = $request.metadata().get("x-forwarded-for") {
```

Comment:
perhaps in the future we can send another x-header that signs the x-forwarded-for in some manner that if both are sent and valid, then we use it.

Reply:
Yeah I'm going to amend this PR so that you can configure the IP source to care about, that way if a node operator has an intermediate worker we will know what to use. There is a complication though, since I believe with proxy protocol, the x-forwarded-for may be the domain name, rather than the resolved IP. Will need to handle both safely in order for this to work (taking whatever raw string may not work as if the proxy is misconfigured, it could add `-` to the x-forwarded-for, as we've seen, which could cause erroneous key collision